swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+
+        self.remove(my_item)
+        self.add(their_item)
+        vendor.add(my_item)
+        vendor.remove(their_item)
+        return True
+
+    def swap_first_item(self, vendor):
+        if not self.inventory or not vendor.inventory:
+            return False
+
+        my_item = self.inventory[0]
+        their_item = vendor.inventory[0]
+
+        self.swap_items(vendor, my_item, their_item)
+        return True
+
+    def get_best_by_category(self, category):
```

Comment:
Awesome implementation: the helper function call, the check for matching_items, and using sort()!